src/bloqade/submission/ir/task_results.py

```diff
@@ -47,15 +60,37 @@ class QuEraTaskStatusCode(str, Enum):
 
 
 class QuEraShotResult(BaseModel):
+    """Object representing results after executing a task in QuEra system.
+
+    Attributes:
+        shot_status (QuEraShotStatusCode): status code of task
+            after running on QuEra system. Defaults to
+            `QuEraShotStatusCode.MissingMeasurement`
+        pre_sequence: Set of preprocessing instructions.
+        post_sequence: Set of preprocessing instructions.
+    """
+
     shot_status: QuEraShotStatusCode = QuEraShotStatusCode.MissingMeasurement
     pre_sequence: conlist(conint(ge=0, le=1), min_items=0) = []
     post_sequence: conlist(conint(ge=0, le=1), min_items=0) = []
 
 
 class TaskProbabilities(BaseModel):
+    """The task results as probabilties.
+
+    Attributes:
+        probabilities (List[Tuple[Tuple[str, str], float]]):
+            task results as probabilities
+    """
+
     probabilities: List[Tuple[Tuple[str, str], float]]
 
-    def simulate_task_results(self, shots=1) -> "QuEraTaskResults":
+    def simulate_task_results(self, shots: int = 1) -> "QuEraTaskResults":
+        """Simulate the task results as probabilties.
+
+        Args:
+            shots (int): Number of shots, Defaults to 1
+        """
```

Comment:
Add some examples and I think this PR should be good to go! sorry for the delay last week was super busy for me.